scripts/o3de/o3de/project_properties.py

```diff
@@ -104,24 +109,43 @@ def edit_project_props(proj_path: pathlib.Path = None,
         proj_json['summary'] = new_summary
     if new_icon:
         proj_json['icon_path'] = new_icon
-    if new_tags:
-        tag_list = new_tags.split() if isinstance(new_tags, str) else new_tags
-        proj_json.setdefault('user_tags', []).extend(tag_list)
-    if delete_tags:
-        removal_list = delete_tags.split() if isinstance(delete_tags, str) else delete_tags
-        if 'user_tags' in proj_json:
-            for tag in removal_list:
-                if tag in proj_json['user_tags']:
-                    proj_json['user_tags'].remove(tag)
-                else:
-                    logger.warning(f'{tag} not found in user_tags for removal.')
-        else:
-            logger.warning(f'user_tags property not found for removal of {delete_tags}.')
-    if replace_tags:
-        tag_list = replace_tags.split() if isinstance(replace_tags, str) else replace_tags
-        proj_json['user_tags'] = tag_list
-    # Update the gem_names field in the project.json
-    _edit_gem_names(proj_json, new_gem_names, delete_gem_names, replace_gem_names, is_optional)
+    if new_version:
+        proj_json['version'] = new_version
+
+    if new_tags or delete_tags or replace_tags != None:
+        proj_json['user_tags'] = utils.update_values_in_key_list(proj_json.get('user_tags', []), new_tags,
+                                                        delete_tags, replace_tags)
+
+
+    if new_gem_names or delete_gem_names or replace_gem_names != None:
+        _edit_gem_names(proj_json, new_gem_names, delete_gem_names, replace_gem_names, is_optional)
+
+
+    def valid_specifier(version_specifier_list):
+        if version_specifier_list and not utils.validate_version_specifier_list(version_specifier_list):
+            logger.error(f'Version specifiers must be in the format <name><version specifiers>. e.g. name==1.2.3 \n {version_specifier_list}')
+            return False
+        return True
```

Comment:
Twice is a trend with this valid_specifier function.
Should it be in validation.py?